WindowsAzureMessaging/WindowsAzureMessaging/MSNotificationHubOptions.m

```diff
@@ -0,0 +1,33 @@
+//----------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//----------------------------------------------------------------
+
+#import <Foundation/Foundation.h>
+#import "MSNotificationHubOptions.h"
+
+@implementation MSNotificationHubOptions
+
+- (instancetype)init {
+    if ((self = [super init])) {
+        if (@available(iOS 10.0, tvOS 10.0, watchOS 3.0, macOS 10.14, macCatalyst 13.0, *)) {
+            _authorizationOptions = (UNAuthorizationOptions)(UNAuthorizationOptionAlert | UNAuthorizationOptionSound | UNAuthorizationOptionBadge);
+        }
+    }
+    
+    return self;
+}
+
+- (instancetype)initWithAuthorizationOptions:(UNAuthorizationOptions)authorizationOptions API_AVAILABLE(ios(10.0), watchos(3.0), macos(10.14), macCatalyst(13.0)) {
+    if ((self = [super init])) {
+        _authorizationOptions = authorizationOptions;
+    }
+    
+    return self;
+}
+
++ (instancetype)optionsWithAuthorizationOptions:(UNAuthorizationOptions)authorizationOptions API_AVAILABLE(ios(10.0), watchos(3.0), macos(10.14), macCatalyst(13.0)) {
```

Comment:
Forgive me - I'm a bit in the dark when it comes to Objective-C, so this question may not make sense.

This looks like a factory method to me, which having the name of a particular dimension baked into it. If that's true, then as we add more dimensions in the future the number of factory methods we need will grow exponentially and get unwieldy with just a few things you can provide.

Would it make more sense to have something like the builder pattern, where you initialize it empty, and add each thing you could care about?

Reply:
Removed the factory method